modules/dxp/apps/portal-workflow/portal-workflow-kaleo-metrics-integration/src/main/java/com/liferay/portal/workflow/kaleo/metrics/integration/internal/helper/IndexerHelper.java

```diff
@@ -124,6 +129,65 @@ public AddNodeRequest createAddNodeRequest(
 		).build();
 	}
 
+	public AddProcessRequest createAddProcessRequest(
+		long companyId, KaleoDefinition kaleoDefinition) {
+
+		AddProcessRequest.Builder builder = new AddProcessRequest.Builder();
+		String version = StringBundler.concat(
+			kaleoDefinition.getVersion(), CharPool.PERIOD, 0);
+
+		builder.active(
+			kaleoDefinition.isActive()
+		).companyId(
+			kaleoDefinition.getCompanyId()
+		).createDate(
+			kaleoDefinition.getCreateDate()
+		).description(
+			kaleoDefinition.getDescription()
+		).modifiedDate(
+			kaleoDefinition.getModifiedDate()
+		).name(
+			kaleoDefinition.getName()
+		).processId(
+			kaleoDefinition.getKaleoDefinitionId()
+		).title(
+			kaleoDefinition.getTitle(
+				LocalizationUtil.getDefaultLanguageId(
+					kaleoDefinition.getTitle()))
+		).titleMap(
+			kaleoDefinition.getTitleMap()
+		).version(
+			version
+		);
+
+		try {
+			List<KaleoDefinitionVersion> kaleoDefinitionVersions =
+				_kaleoDefinitionVersionLocalService.getKaleoDefinitionVersions(
+					companyId, kaleoDefinition.getName());
+
+			if (kaleoDefinitionVersions != null) {
+				return builder.versions(
+					Stream.of(
+						kaleoDefinitionVersions
+					).flatMap(
+						List::stream
+					).map(
+						KaleoDefinitionVersion::getVersion
+					).toArray(
+						String[]::new
+					)
+				).build();
+			}
+		}
+		catch (PortalException portalException) {
+			portalException.printStackTrace();
```

Comment:
Hi @seltong, use `_log.warn...` instead `portalException.printStackTrace();` and before to log a warn check if warn is enabled `_log.isWarnEnabled()`

Thank you!